data_loader.py

```diff
@@ -265,7 +256,10 @@ def validate_files(self, cheat_mode, loading_mode, file_list, max_violations, te
             return True
 
     def load(self, file_list, cheat_mode, dry_run, loading_mode, wipe_db, max_violations, temp_folder, verbose,
-             split=False, no_backup=True, backup_folder="/", neo4j_uri=None):
+             split=False, no_backup=True, neo4j_uri=None, backup_folder="/", username=None, password=None):
+        if self.database_type not in self.allowed_database_type:
```

Comment:
This check should happen when gathering configuration in process_arguments function for example.